File: 15_drag_nestedscroll/src/main/java/one/diao/com/a15_drag_nestedscroll/MainActivity.java
package one.diao.com.a15_drag_nestedscroll;

import android.os.SystemClock;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.drag_listener_grid_view);

        SystemClock.sleep(2000);
    }
}

File: constraintlayout/src/main/java/com/one/constraintlayout/MainActivity.kt
package com.one.constraintlayout

import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.LinearLayout

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)


     val linearLayout =    findViewById<LinearLayout>(R.id.root)
        packageManager.getPackageInfo(packageName, PackageManager.GET_ACTIVITIES).activities.forEach {
            activity ->
            if (activity.name == this::class.java.name) {
                return@forEach
            }

            val clazz = Class.forName(activity.name)

            val button = Button(this).apply {
                isAllCaps = false
                text = clazz.simpleName

                setOnClickListener {
                    startActivity(Intent(this@MainActivity,clazz))
                }

            }

            linearLayout.addView(button)



        }


    }
}
